fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneRecordCursor.java

```diff
@@ -289,10 +183,16 @@ private synchronized IndexReader getIndexReader() throws IOException {
     }
 
     private void performScan() throws IOException {
+        if (lookupResults != null && currentPosition < pageSize) {
```

Comment:
Should there be modular arithmetic here? Something like `currentPosition % pageSize != 0`?

Reply:
Oh, I see--I think this is fine